src/v/redpanda/admin_server.cc

```diff
@@ -3339,3 +3346,45 @@ void admin_server::register_shadow_indexing_routes() {
           return sync_local_state_handler(std::move(req));
       });
 }
+
+ss::future<> admin_server::restart_redpanda_service(service_kind service) {
+    if (service == service_kind::schema_registry) {
+        // Checks specific to schema registry
+        if (_schema_registry == nullptr) {
+            throw ss::httpd::server_error_exception(
+              "Schema Registry is undefined. Is it set in the .yaml config "
+              "file?");
+        }
+
+        try {
+            co_await _schema_registry->restart();
+        } catch (...) {
+            throw ss::httpd::server_error_exception(
+              "Unknown issue restarting schema_registry");
```

Comment:
nit: log additional context to make debugging easier in the future. for example here all of the context in the exception is thrown away. no need to return all that context to the user, but logging it is useful.